### PR TITLE
Use NDArithmeticMixin methods if astropy >= 1.2

### PR DESCRIPTION
**Pull request by MSeifert04**
Please have a look at the following list and replace the "[ ]" with a "[x]" if
the answer to this question is yes.

- [ ] For new contributors: Did you add yourself to the "Authors.rst" file?

For documentation changes:

- [ ] For documentation changes: Does your commit message include a "[skip ci]"?
      Note that it should not if you changed any examples!

For bugfixes:

- [ ] Did you add an entry to the "Changes.rst" file?
- [ ] Did you add a regression test?
- [ ] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Does this PR add, rename, move or remove any existing functions or parameters?

For new functionality:

- [ ] Did you add an entry to the "Changes.rst" file?
- [ ] Did you include a meaningful docstring with Parameters, Returns and Examples?
- [ ] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Did you include tests for the new functionality?
- [x] Does this PR add, rename, move or remove any existing functions or parameters?

Please note that the last point is not a requirement. It is meant as a check if
the pull request potentially breaks backwards-compatibility.

-----------------------------------------

This is currently work-in-progress. But since the astropy 1.2 release I wanted to give this a shot.

It uses the `NDArithmeticMixin` methods for add/subtract/multiply/divide, which allows to use CCDData in arithmetic operations with numpy-arrays, scalars and some other objects. For this to work I needed to remove the mandatory `CCDData` unit - because `NDArithmeticMixin` converts the second operator to a `CCDData` which fails if it's a numpy.ndarray or scalar.

### CODE REVIEW

**Comment by MSeifert04** [comment]
Related to (and maybe fixes) #326

**Comment by MSeifert04** [comment]
and if the `ValueError` shouldn't be removed when no `unit` is given let me know. That would require hacking into the `NDArithmeticMixin` internals.

**Comment by mwcraig** [comment]
My initial reaction is that it would be nice to preserve the unit requirement, and that perhaps it is a change I'd only want to make in a backwards-incompatible release like 2.0. It isn't backwards-incompatible in the *code* sense, but the *behavior* would be quite a bit different than what it has been in the past.

~~Is there any way to pass the unit information in to the mixin (I honestly do not remember the details). If not, might it be possible to infer the units of the result and add them afterwards?~~

Sorry, just understood what the issue is...will think some more about workarounds.

**Comment by MSeifert04** [comment]
It's not complicated to preserve the unit requirement and just _temporarly_ disable it when the `NDArithmeticMixin` runs (and tries to convert everything to `CCDData`). If that would be better I'll adjust the PR.

**Comment by mwcraig** [comment]
Yes, please! 😀

**Comment by MSeifert04** [comment]
In the end I used a global variable to temporarly disable the unit-checks. I thought about using a contextmanager or a global dictionary but in the end that's much more effort for just one trigger. But I realize that this might not be optimal.

I'll add some tests later.

File: ccdproc/core.py [hunk 1]
```diff
@@ -703,6 +703,7 @@ def gain_correct(ccd, gain, gain_unit=None):
         gain_value = gain
 
     result = ccd.multiply(gain_value)
+    result.meta = ccd.meta.copy()
```
Comment (mwcraig) [inline ccdproc/core.py]: This is also unrelated to the broader change, correct? (agreed it should be fixed) 

Is there a regression test we should add here?

File: .travis.yml [hunk 2]
```diff
@@ -55,6 +55,7 @@ matrix:
         # Check compatibility with the current astropy LTS release
         - python: 2.7
           env: ASTROPY_VERSION=LTS
+               SETUP_CMD='test --coverage'
```
Comment (MSeifert04) [inline .travis.yml]: More to have a coverage of the LTS branch, otherwise either the coverage decreases significantly or I had to put a `# pragma no cover` after each of the arithmetic function.

File: ccdproc/core.py (hunk 1 above)
Comment (MSeifert04) [inline ccdproc/core.py]: Now that you mention it, I investigated a bit and found why this change was necessary:

The `_ccddata_arithmetic` makes sure the meta is kept when the other operand is a scalar (not CCDData), the `NDArithmeticMixin` however defaults (because of backwards-compatibility-reasons) to no discarding the `meta`. 
Note that currently operations between two CCDData independant of the astropy version discard their meta, it was just scalars that had the special case.

Comment (MSeifert04) [inline ccdproc/core.py]: But I don't know if it's worth keeping the meta only if and only if the other operand has none. That's inconsistent with the case that the other operand is a CCDData, then the meta is always discarded.

**Comment by mwcraig** [comment]
Any idea what is causing the test fail? Doesn't seem related to this at all.

**Comment by mwcraig** [comment]
I think (despite the line numbers reported in the test failure) that this is maybe because the add, etc, are no longer methods?

**Comment by mwcraig** [comment]
Code-wise I like the last commit (less repetition is better), not sure how to fix the references in the documentation, though...

**Comment by MSeifert04** [comment]
I changed it so it's not using `functools.wraps` anymore and write the docstring manually. At least locally that fixed sphinx and did not fail the tests.

**Comment by MSeifert04** [comment]
I guess it was the lazy-loading PR (https://github.com/astropy/astropy/pull/5065) merge which breaks the tests against astropy-dev.

**Comment by MSeifert04** [comment]
The astropy-dev failure bugfix is in #435 so this will fail the corresponding travis test. But the test before the "bugfix" reversal passed.

**Comment by mwcraig** [comment]
Merging, will check the builds in the morning after merging.
